src/c++/perf_analyzer/infer_data_manager_shm.cc

```diff
@@ -126,8 +127,17 @@ InferDataManagerShm::Init()
           }
 
           RETURN_IF_ERROR(data_loader_->GetInputData(
-              input.second, i, (j + count) % data_loader_->GetTotalSteps(i),
+              tensor, i, (j + count) % data_loader_->GetTotalSteps(i),
               &data_ptr, &batch1_bytesize));
+
+          // FIXME: TMA-765 - Shared memory mode does not support optional
+          // inputs, currently, and will be implemented in the associated story.
+          if (data_ptr == nullptr) {
+            return cb::Error(
```

Comment:
What happened before this code? A less-descriptive error?

Reply:
I'm pretty sure it would have seg-faulted trying to copy data from a nullptr, but I haven't confirmed